pkg/skaffold/event/event.go

```diff
@@ -677,6 +696,17 @@ func ResetStateOnBuild() {
 	handler.setState(newState)
 }
 
+// ResetStateOnTest resets the test, deploy, sync and status check state
+func ResetStateOnTest() {
+	newState := handler.getState()
+	newState.TestState.Status = NotStarted
+	newState.TestState.StatusCode = proto.StatusCode_OK
+	newState.StatusCheckState = emptyStatusCheckState()
+	newState.ForwardedPorts = map[int32]*proto.PortEvent{}
+	newState.DebuggingContainers = nil
```

Comment:
I don't think `Test` phase should reset these fields since it by itself doesn't trigger any redeployment, neither does it make the existing deployment stale.

Reply:
Removed the lines.